src/MenuElementInterface.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace RebelCode\WordPress\Admin\Menu;
+
+use Dhii\Data\IdAwareInterface;
+
+/**
+ * Something that represents an element that can be placed into a menu or behave as a menu.
+ *
+ * @since [*next-version*]
+ */
+interface MenuElementInterface extends IdAwareInterface
```

Comment:
Consider using the new `dhii/data-tree-interface`. Be advised: it uses key instead of ID.

[`dhii/data-tree-interface`]:                                     https://github.com/Dhii/data-tree-interface